mt_dnn/batcher.py

```diff
@@ -34,6 +53,27 @@ def _get_shuffled_index_batches(dataset_len, batch_size):
         random.shuffle(index_batches)
         return index_batches
 
+    @staticmethod
+    def _get_shuffled_index_batches_bin(dataset, batch_size, bin_size, bin_grow_ratio):
+        maxlen = dataset._maxlen
+        bins = create_bins(bin_size, maxlen)
+        data = [[] for i in range(0, len(bins))]
+        
+        for idx, sample in enumerate(dataset):
+            bin_idx = search_bin(bins, len(sample['sample']['token_id']))
+            data[bin_idx].append(idx)
+        index_batches = []
+
+        for idx, sub_data in enumerate(data):
+            if len(sub_data) < 1: continue
+            batch_size = int(batch_size * bin_grow_ratio)
```

Comment:
from current code, the batch size of shortest seq is int(batch_size * bin_grow_ratio) instead of batch_size, this has 2 disadvantages:
1. confusing.
2. even the shortest seq batch (largest batch) 's loss is scaled down with code in model.py "loss = loss * (1.0 * batch_size / self.config['batch_size'])"

better to move line 69-70 after line 73

Reply:
thanks, fixed.